e2e/nsenter.go

```diff
@@ -134,6 +122,32 @@ func extractClusterParameters(ctx context.Context, t *testing.T, kube *kubeclien
 	return result, nil
 }
 
+// Returns the name of a pod that's a member of the 'debug' daemonset, running on an aks-nodepool node
+func getDebugPodName(kube *kubeclient) (string, error) {
+	podList := corev1.PodList{}
+	if err := kube.dynamic.List(context.Background(), &podList, client.MatchingLabels{"app": "debug"}); err != nil {
+		return "", fmt.Errorf("failed to list debug pod: %q", err)
+	}
+
+	if len(podList.Items) < 1 {
+		return "", fmt.Errorf("failed to find debug pod, list by selector returned no results")
+	}
+
+	var podName string
+	for _, pod := range podList.Items {
+		if strings.Contains(pod.Spec.NodeName, "aks-nodepool") {
```

Comment:
we should only be using the debug pod present on the original aks nodepool's single node, rather than potentially using a pod on an ephemeral node from other testing happening simultaneously

was also thinking it might be helpful to bump the number of nodes in the original nodepool to 2 or 3 to make high volumes of parallel tests more robust